### PR TITLE
Add TogglesClient and TogglesAwareTestRule

### PR DESCRIPTION
**Pull request by volteanu**

### CODE REVIEW

File: src/main/java/com/adobe/cq/testing/junit/rules/toggles/TogglesAwareTestRule.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        }
+
+        if (runIfToggle != null && !enabledToggles.get().contains(runIfToggle)) {
+            return false; // RunIfToggleEnabled condition was not met, skipping
+        }
+
+        return skipIfToggle == null || !enabledToggles.get().contains(skipIfToggle);
+    }
+
+    private String getRunIfToggle(Description description) {
```
Comment (dulvac) [inline src/main/java/com/adobe/cq/testing/junit/rules/toggles/TogglesAwareTestRule.java]: minor: should we make this an `Optional<String>` ?

**Review by dulvac — APPROVED** [review]
small question around the get*Toggle methods: Do we want to make that return an optional?since this seems to me the classic case for Optionals

**Review by marcpf — APPROVED** [review]
LGTM.

File: src/main/java/com/adobe/cq/testing/junit/rules/toggles/TogglesAwareTestRule.java (hunk 1 above)
Comment (volteanu) [reply src/main/java/com/adobe/cq/testing/junit/rules/toggles/TogglesAwareTestRule.java]: 👍 I've changed that in 03b7fa16517690ea71ef76b879c89ce02d728104. WDYT?

**Pull request merged**